memflow/src/plugins/os/system.rs

```diff
@@ -0,0 +1,243 @@
+use crate::error::*;
+
+use super::{ArcPluginProcess, OSLayerFunctionTable, PluginProcess};
+use crate::os::{AddressCallback, ModuleInfo, OSInfo, OSInner, Process, ProcessInfo};
+use crate::types::Address;
+use std::ffi::c_void;
+
+use super::super::COptArc;
+use super::PluginOS;
+use super::{MUArcPluginProcess, MUModuleInfo, MUPluginProcess, MUProcessInfo};
+
+use libloading::Library;
+
+pub type OpaqueOSFunctionTable = OSFunctionTable<'static, c_void, c_void>;
+
+impl Copy for OpaqueOSFunctionTable {}
+
+impl Clone for OpaqueOSFunctionTable {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+#[repr(C)]
+pub struct OSFunctionTable<'a, P, T> {
```

Comment:
The Plugin API is still missing support for Keyboard-Input extensions (that the win32 ffi exposes). We should add a wrapper that just takes a `int` type for each key. This will work with win32 and linux.